recycleIT-frontend/src/Components/Services/index.tsx

```diff
@@ -1,13 +1,68 @@
-import Header from '../Header/Header'
-import Map from './Map/Map'
-import ServicesList from './ServicesList/ServicesList'
+import Header from '../Header/Header';
+import Map from './Map/Map';
+import ServicesList from './ServicesList/ServicesList';
+import { useState, useEffect, createContext } from 'react'
+import useMediaQuery from '@mui/material/useMediaQuery';
+import { 
+    LocationType, 
+    ChosenItemType, 
+    IServiceList 
+} from '../interfaces/Interfaces';
+
+export const UserLocation = createContext<LocationType | null>(null);
 
 const Services = () => {
+
+    const [servicesList, setServicesList] = useState<IServiceList[] | null>(null);
+    const [isDrawerOpened, setIsDrawerOpened] = useState(false);
+    const [userLocation, setUserLocation] = useState<LocationType | null>(null);
+    const [chosenItemLocation, setChosenItemLocation] = useState<ChosenItemType | null>(null);
+    const isMobileDevice = useMediaQuery('(min-width:900px)');
+
+    useEffect(() => {
+        // get user`s current location
+        navigator.geolocation.getCurrentPosition((position) => {
+            console.log('success', position);
+            setUserLocation({lat: position.coords.latitude, lng: position.coords.longitude});
+        });
+    }, []);
+
+    useEffect(() => {
+        setIsDrawerOpened(isMobileDevice);
+    }, [isMobileDevice])
+
+    const setServicesData = (services: IServiceList[] | null) => {
+        console.log('service list updated');
+        setServicesList(services);
+    }
+
+    const setItemLocation = (id: number, lat: number, lng: number) => {
+        setChosenItemLocation({id: id, lat: lat, lng: lng});
+    }
+
+    const toggleDrawerOpened = () => {
+        setIsDrawerOpened(prev => !prev)
+    }
+
     return (
         <>
```

Comment:
It's a good idea to provide some sort of wrapper component and make sure that warpper defines layout of your inner components. This will allow for reusability of any child component if you need that in the future (e.g. place map on some other page)

Reply:
Do I understand correctly that the idea is in our case to create a component that would simply have the layout of sidebar on the left and main content on the right? And this component will take care of toggling the sidebar, so that we do not pass such props as toggleDrawerOpened, isDrawerOpened etc., which are, in fact, not necessary for the Map and ServicesList components. And inside that sidebar and content, we will be able to place any component that we want. I have tried to implement that and for now this wrapper looks like that:
`<ServicesWrapper 
                    isDrawerOpened={isDrawerOpened} 
                    isMobileDevice={isMobileDevice}
                    toggleDrawerOpened={toggleDrawerOpened}
                    heading='Search for eco services, available for you'
                    side={<ServicesList 
                        servicesList={servicesList} 
                        setServicesList={setServicesData}
                        setItemLocation={setItemLocation}
                    />}
                    content={<Map 
                        servicesList={servicesList}
                        chosenItemLocation={chosenItemLocation}
                    />}
                />`

Reply:
That seems like a bit of an overkill. I meant a simple div with apllied styles instead for <> </> that would take care about child components positioning. Option you provided would be super useful in case you have a lot of similarly structured layouts that could reuse that logic, but in our case a regular div will do :)

Reply:
Okey, now I get that